File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/companies/model/CompanyAlternativeName.java
package dev.reuss.tmdb.domain.companies.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Alternative name for a TMDB company.
 *
 * @param name alternative company name
 * @param type optional name type
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record CompanyAlternativeName(
        String name,
        String type
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/companies/model/CompanyAlternativeName.kt
package dev.reuss.tmdb.domain.companies.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Alternative name for a TMDB company.
 *
 * @property name alternative company name
 * @property type optional name type
 */
@JvmRecord
data class CompanyAlternativeName(
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("type")
    val type: String?,
) : TmdbModel
